sklearn/decomposition/_base.py

```diff
@@ -124,7 +123,7 @@ def transform(self, X):
         """
         check_is_fitted(self)
 
-        X = check_array(X)
+        X = self._validate_data(X, dtype=[np.float64, np.float32], reset=False)
```

Comment:
do we need to pass the dtype here?

Reply:
I see no reason why this could would not work properly in float32 so this is a slight performance improvement.

Reply:
Note that the fit of PCA accepts float32 without upcasting so this change makes transform consistent with fit.